File: com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/IInterstitialEventListener.java
package com.chartboost.heliumsdk.unity;

/**
 * Interface for sending real-time Interstitial events
 */
public interface IInterstitialEventListener {
    void DidLoadInterstitial(String placementName, int errorCode, String errorDescription);
    void DidClickInterstitial(String placementName, int errorCode, String errorDescription);
    void DidCloseInterstitial(String placementName, int errorCode, String errorDescription);
    void DidShowInterstitial(String placementName, int errorCode, String errorDescription);
    void DidRecordImpression(String placementName, int errorCode, String errorDescription);
    void DidWinBidInterstitial(String placementName, String auctionId, String partnerId, double price);
}

File: com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/IInterstitialEventListener.kt
package com.chartboost.heliumsdk.unity

/**
 * Interface for sending real-time Interstitial events
 */
interface IInterstitialEventListener {
    fun DidLoadInterstitial(placementName: String, loadId: String, error: String)
    fun DidShowInterstitial(placementName: String, error: String)
    fun DidCloseInterstitial(placementName: String, error: String)
    fun DidClickInterstitial(placementName: String)
    fun DidRecordImpression(placementName: String)
    fun DidReceiveReward(placementName: String)
    fun DidWinBidInterstitial(placementName: String, auctionId: String, partnerId: String, price: Double, error: String)
}
